File: python-gdb-activities-20/activity17/gdb/command/withdraw_command.py
# gdb/command/withdraw_command.py
from typing import Optional
from gdb.domain.iaccount import IAccount
from gdb.domain.transaction import Transaction
from gdb.command.transaction_command import TransactionCommand

class WithdrawCommand(TransactionCommand):
    """Command executing a withdrawal operation."""

    # ============================================================
    # 📝 STEP 5: Constructor
    #
    # INSTRUCTIONS:
    #   1. Store account in self._account.
    #   2. Store amount (as a float) in self._amount and pin (as an int) in self._pin.
    #   3. Set self._transaction = None -- execute() fills it in later.
    # ============================================================
    # TODO: store the account, amount and PIN, and start with no transaction
    def __init__(self, account: IAccount, amount: float, pin: int) -> None:
        raise NotImplementedError("TODO: Step 5 - implement WithdrawCommand constructor")

    # ============================================================
    # 📝 STEP 6: Implement execute()
    #
    # INSTRUCTIONS:
    #   Call account.withdraw_with_transaction(amount, pin) and store in self._transaction.
    # ============================================================
    # TODO: execute withdrawal
    def execute(self) -> None:
        pass

    # ============================================================
    # 📝 STEP 7: Implement get_transaction()
    #
    # INSTRUCTIONS:
    #   Return self._transaction (it stays None until execute() has run).
    # ============================================================
    # TODO: return _transaction
    def get_transaction(self) -> Optional[Transaction]:
        raise NotImplementedError("TODO: Step 7 - implement get_transaction")

File: java-gdb-activities-20/activity17/src/com/gdb/command/WithdrawCommand.java
package com.gdb.command;

import com.gdb.domain.Account;
import com.gdb.domain.IAccount;
import com.gdb.domain.Transaction;

public class WithdrawCommand implements TransactionCommand {
    private static final long serialVersionUID = 1L;

    // ============================================================
    // 📝 STEP 3: Declare Fields
    //
    // INSTRUCTIONS:
    //   1. private IAccount account;
    //   2. private double amount;
    //   3. private int pin;
    //   4. private Transaction transaction;
    // ============================================================
    // TODO: declare account, amount, pin, and transaction fields

    // ============================================================
    // 📝 STEP 4: Write Constructor
    //
    // INSTRUCTIONS:
    //   Accept (IAccount account, double amount, int pin); store all.
    // ============================================================
    // TODO: implement constructor
    public WithdrawCommand(IAccount account, double amount, int pin) {
        // TODO: Step 4 - implement constructor
    }

    // ============================================================
    // 📝 STEP 5: Implement execute()
    //
    // INSTRUCTIONS:
    //   1. Call account.withdrawWithTransaction(amount, pin).
    //   2. Store returned Transaction in this.transaction.
    // ============================================================
    // TODO: implement execute() method
    @Override
    public void execute() throws Exception {
        // TODO: Step 5 - implement execute()
    }

    // ============================================================
    // 📝 STEP 6: Implement getTransaction()
    //
    // INSTRUCTIONS:
    //   Return this.transaction.
    // ============================================================
    // TODO: return this.transaction
    @Override
    public Transaction getTransaction() {
        // TODO: Step 6 - return this.transaction
        return null;
    }
}
